mbuild/formats/hoomdxml.py

```diff
@@ -88,157 +71,239 @@ def write_hoomdxml(structure, filename, box, ref_distance=1.0, ref_mass=1.0,
+    ref_energy : float, default=1.0
+        Reference energy for conversion to reduced units
+
+    """
+
+    unique_bond_types = set()
+    xml_file.write('<bond>\n')
+    for bond in structure.bonds:
+        t1, t2 = bond.atom1.type, bond.atom2.type
+        if t1 == '' or t2 == '':
+            t1, t2 = bond.atom1.name, bond.atom2.name
+        t1, t2 = sorted([t1, t2])
+        try:
+            bond_type = ('-'.join((t1, t2)), bond.type.k, bond.type.req)
+        except AttributeError:  # no forcefield applied, bond.type is None
+            bond_type = ('-'.join((t1, t2)), 0.0, 0.0)
+        unique_bond_types.add(bond_type)
+        xml_file.write('{} {} {}\n'.format(
+            bond_type[0], bond.atom1.idx, bond.atom2.idx))
+    xml_file.write('</bond>\n')
+    xml_file.write('<bond_coeffs>\n')
+    xml_file.write('<!-- type k r_eq -->\n')
+    for bond_type, k, req in unique_bond_types:
+        xml_file.write('{} {} {}\n'.format(bond_type,
+            k * 2.0 / ref_energy * ref_distance**2.0, req/ref_distance))
+    xml_file.write('</bond_coeffs>\n')
+
+
+def _write_angle_information(xml_file, structure, ref_energy):
+    """Write the angles in the system.
+
+    Parameters
+    ----------
+    xml_file : file object
+        The file object of the hoomdxml file being written
+    structure : parmed.Structure
+        Parmed structure object
+    ref_energy : float, default=1.0
+        Reference energy for conversion to reduced units
+
+    """
+
+    unique_angle_types = set()
+    xml_file.write('<angle>\n')
+    for angle in structure.angles:
+        t1, t2, t3 = angle.atom1.type, angle.atom2.type, angle.atom3.type
+        t1, t3 = sorted([t1, t3])
+        angle_type = ('-'.join((t1, t2, t3)), angle.type.k, angle.type.theteq)
+        unique_angle_types.add(angle_type)
+        xml_file.write('{} {} {} {}\n'.format(
+            angle_type[0], angle.atom1.idx, angle.atom2.idx, angle.atom3.idx))
+    xml_file.write('</angle>\n')
+    xml_file.write('<angle_coeffs>\n')
+    xml_file.write('<!-- type k theta_eq -->\n')
+    for angle_type, k, teq in unique_angle_types:
+        xml_file.write('{} {} {}\n'.format(angle_type,
+            k * 2.0 / ref_energy, radians(teq)))
+    xml_file.write('</angle_coeffs>\n')
+
+
+def _write_dihedral_information(xml_file, structure, ref_energy):
+    """Write dihedrals in the system.
+
+    Parameters
+    ----------
+    xml_file : file object
+        The file object of the hoomdxml file being written
+    structure : parmed.Structure
+        Parmed structure object
+    ref_energy : float, default=1.0
+        Reference energy for conversion to reduced units
+
+    """
+
+    unique_dihedral_types = set()
+    xml_file.write('<dihderal>\n')
+    for dihedral in structure.rb_torsions:
+        t1, t2 = dihedral.atom1.type, dihedral.atom2.type,
+        t3, t4 = dihedral.atom3.type, dihedral.atom4.type
+        st2, st3 = sorted([t2, t3])  # sort based on the middle 2
```

Comment:
It looks like this line can be removed since you don't end up using these variables